test/integration/Main.hs

```diff
@@ -43,66 +35,94 @@ main = do
         describe "Cardano.WalletSpec" Wallet.spec
         describe "Cardano.Wallet.Network.HttpBridge" HttpBridge.spec
 
-        beforeAll (withWallet (newMVar . Context ())) $ do
+        beforeAll startCluster $ afterAll killCluster $ do
             describe "Integration test framework" dummySpec
 
-        beforeAll (dummySetup (newMVar . Context ())) $ do
+        beforeAll dummySetup $ do
             describe "Test response codes" respCodesSpec
+  where
+    -- Run a local cluster of cardano-sl nodes, a cardano-http-bridge on top and
+    -- a cardano wallet server connected to the bridge.
+    startCluster :: IO Context
+    startCluster = do
+        let stateDir = "./test/data/cardano-node-simple"
+        systemStart <-
+            formatTime defaultTimeLocale "%s" . addUTCTime 10 <$> getCurrentTime
+        cluster <- async $ void $ launch
+            [ cardanoNodeSimple stateDir systemStart ("core0", "127.0.0.1:3000")
+            , cardanoNodeSimple stateDir systemStart ("core1", "127.0.0.1:3001")
+            , cardanoNodeSimple stateDir systemStart ("core2", "127.0.0.1:3002")
+            , cardanoNodeSimple stateDir systemStart ("relay", "127.0.0.1:3100")
+            , cardanoHttpBridge "8080" "local"
+            , cardanoWalletServer "1337" "8080" "local"
+            ]
+        link cluster
+        let baseURL = "http://localhost:1337/"
+        manager <- newManager defaultManagerSettings
+        return $ Context cluster (baseURL, manager)
+
+    killCluster :: Context -> IO ()
+    killCluster (Context cluster _) = cancel cluster
+
+    cardanoNodeSimple stateDir systemStart (nodeId, nodeAddr) = Command
+        "cardano-node-simple"
+        [ "--system-start", systemStart
+        , "--node-id", nodeId
+        , "--keyfile", stateDir <> "/keys/" <> nodeId <> ".sk"
+        , "--configuration-file", stateDir <> "/configuration.yaml"
+        , "--configuration-key", "default"
+        , "--topology", stateDir <> "/topology.json"
+        , "--db-path", "/tmp/cardano-node-simple/db/" <> nodeId
+        , "--listen", nodeAddr
+        , "--log-config", stateDir <> "/logs/" <> nodeId <> "/config.json"
+        , "--rebuild-db"
+        ] (pure ())
+
+    cardanoHttpBridge port network = Command
+        "cardano-http-bridge"
+        [ "start"
+        , "--port", port
+        , "--template", network
+        ] (threadDelay 5000000)
+
+    cardanoWalletServer serverPort bridgePort network = Command
```

Comment:
Why not `cardanoWalletLauncher` instead `Bridge` and `Server`?

Reply:
Good point :no_mouth:  ... I have no answers.